build-logic/src/main/kotlin/org/jetbrains/conventions/kotlin-jvm.gradle.kts

```diff
@@ -10,25 +11,44 @@ plugins {
 
 configureDokkaVersion()
 
-val projectsWithoutOptInDependency = setOf(
-    ":integration-tests", ":integration-tests:gradle", ":integration-tests:maven", ":integration-tests:cli")
+kotlin {
+    explicitApi = ExplicitApiMode.Strict
```

Comment:
If `explicitApi` was set in `tasks.withType<KotlinCompile>()`, it would be applied to the test sources as well, which we don't want. But setting it in `kotlin {}` fixes it, so I did that and moved other compiler arguments into it as well.